internal/gatherer/gatherer_test.go

```diff
@@ -252,15 +170,164 @@ func TestProcessNamespace(t *testing.T) {
 			hasMetricsAPI:  true,
 			expectedNsInfo: &models.NamespaceInfo{
 				Pods:            0,
-				IsIstioInjected: true,
+				IsIstioInjected: false, // this is false because no pods within the namespace have istio sidecar injected
```

Comment:
I'm a bit confused how this agrees with comments like "If the namespace has istio injection enabled, or if the pod has istio injection enabled, we should count it as istio injected".

Reply:
I think a breaking change is probably not worth it - you can just clarify the meaning with comments.